src/collectives.c

```diff
@@ -613,8 +613,19 @@ shmem_internal_op_to_all_linear(void *target, const void *source, size_t count,
         SHMEM_WAIT_UNTIL(pSync, SHMEM_CMP_EQ, 0);
 
         /* send data, ack, and wait for completion */
+#ifdef DISABLE_NONFETCH_AMO
```

Comment:
Is it possible these patches could go into `shmem_internal_atomicv`?  If so, it seems like that might be better "layering".

Reply:
Sure! I've made the recommended change, though it does require reacquiring the **type_size** and **count** variables from the **len** parameter passed into **shmem_internal_atomicv**, which does require a division operation.

Reply:
Thanks @philipmarshall21.  I'd say the extra division probably isn't too bad considering the ~1/2 round-trip-time added by this workaround (which is hopefully temporary anyway).